projects/dotcms-field-elements/src/components/dot-textfield/dot-texfield.e2e.ts

```diff
@@ -1,11 +1,130 @@
-  });
+        spyStatusChangeEvent = await page.spyOnEvent('statusChanges');
+        spyValueChanges = await page.spyOnEvent('valueChanges');
+        element = await page.find('dot-textfield');
+        input = await page.find('input');
+    });
+
+    it('should render', async () => {
+        const tagsRenderExpected = `<label>Name:</label><input name="fullName" type="text" placeholder="Enter Name" required=""><span class="dot-textfield__hint">this is a hint</span>`;
+        expect(element.innerHTML).toBe(tagsRenderExpected);
+    });
+
+    it('should show Regex validation message', async () => {
+        await input.press('@');
+        await page.waitForChanges();
+        const errorMessage = await page.find('.dot-textfield__error-meessage');
+        expect(errorMessage.innerHTML).toBe('Invalid Name');
+    });
+
+    describe('emit statusChanges', () => {
+        //TODO: In progress...
+        it('should send status onBlur', async () => {
+            await page.$eval('input', (e: HTMLElement) => e.blur());
+            await page.waitForChanges();
+            expect(spyStatusChangeEvent).toHaveReceivedEventDetail({
+                name: 'fullName',
+                dotPristine: false,
+                dotTouched: true,
+                dotValid: true
+            });
+        });
+
+        it('should send status value change', async () => {
+            input.press('a');
+            await page.waitForChanges();
+            expect(spyStatusChangeEvent).toHaveReceivedEventDetail({
+                name: 'fullName',
+                dotPristine: false,
+                dotTouched: true,
+                dotValid: true
+            });
+        });
+
+        it('should send status on Reset', async () => {
+            element.callMethod('reset');
+            await page.waitForChanges();
+            expect(spyStatusChangeEvent).toHaveReceivedEventDetail({
+                name: 'fullName',
+                dotPristine: true,
+                dotTouched: false,
+                dotValid: false
+            });
+        });
+    });
+
+    it('should emit change value', async () => {
+        input.press('a');
+        await page.waitForChanges();
+        expect(spyValueChanges).toHaveReceivedEventDetail({ name: 'fullName', value: 'Johna' });
+    });
+
+    it('should load as pristine and untouched', async () => {
+        expect(element.classList.contains('dot-pristine')).toBeTruthy();
+        expect(element.classList.contains('dot-untouched')).toBeTruthy();
+    });
+
+    //TODO: In progress...
+    it('should mark as touched when onblur', async () => {
+        // input.press('@');
+        await page.waitForChanges();
+        expect(element.classList.contains('dot-touched')).toBeTruthy();
+    });
+
+    it('should mark as dirty and touched when type', async () => {
+        input.press('a');
+        await page.waitForChanges();
+        expect(element.classList.contains('dot-dirty')).toBeTruthy();
```

Comment:
`classList.contains` returns `true/false` is not a truthy, so this assert should be `toBe(true/false)`